### PR TITLE
Adding ability for SQS source CRD to use annotations

### LINKED ISSUE #993: SQS Source: time.Time: not in RFC3339 format
**Describe the bug**
When trying to send to in-memory channel or another knative service from SQS Source I receive the following error.

{"level":"error","ts":1583510808.2513373,"logger":"fallback","caller":"http/transport.go:572","msg":"failed to convert http message to event","error":"cannot convert \"52149-06-15T07:22:46Z\" to time.Time: not in RFC3339 format","stacktrace":"knative.dev/eventing-contrib/vendor/github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http.(*Transport).ServeHTTP\n\t/home/prow/go/src/knative.dev/eventing-contrib/vendor/github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http/transport.go:572\nnet/http.(*ServeMux).ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2387\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2802\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:1890"}

 **Expected behavior**
I expect the cloud event to be successfully transmitted to the sink

 **To Reproduce**
Create Knative service
```
apiVersion: serving.knative.dev/v1alpha1
kind: Service
metadata:
  name: awssqs-event-display
  namespace: content-platform-sqs-source
spec:
  runLatest:
    configuration:
      revisionTemplate:
        spec:
          container:
            image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display@sha256:b2c341b2f36c46ed20041364e19cdaa19c7002dad97646f69717257b5afa4512
```

Create Knative SQS event source
```
apiVersion: sources.eventing.knative.dev/v1alpha1
kind: AwsSqsSource
metadata:
  name: awssqs-sample-source
  namespace: content-platform-sqs-source
spec:
  awsCredsSecret:
    name: aws-credentials
    key: credentials
  queueUrl: "https://sqs.us-east-1.amazonaws.com/xxxx/sqs-test"
  sink:
    apiVersion: serving.knative.dev/v1alpha1
    kind: Service
    name: awssqs-event-display
```

push sqs event
Either command line or UI

Error now happens

 **Knative release version**
Serving: v0.11.1
Eventing: v0.11.0
SQS Source v0.11.2

 **Additional context**
We are in EKS so we can only use Kubernetes 1.14 for now, so I would really like to see a fix in 0.11.x since we can't upgrade to 0.12.x yet because of EKS.

This is a blocking issue for us currently.

Thanks.

### PR DESCRIPTION
**Pull request by AceHack**
Fixes #993 #986 #1012

Backporting:
#996
#1022
#1013

## Proposed Changes

Added the ability for AwsSqsSource CRD to expose pod annotations, this will allow for using Kube2IAM credentials. The existing credentials file is also supported. This is a backward-compatible change.

Fixing issue with incorrect conversion of sqs SentTimestamp

🎁 Add new feature
🐛 Fix bug

**Release Note**

```release-note
Added the ability for AwsSqsSource CRD to expose pod annotations, this will allow for using Kube2IAM credentials.  The existing credentials file is also supported.
```

**Docs**
awssqs/samples/README.md was updated as part of this PR

### CODE REVIEW

File: awssqs/samples/awssqs-source-kube2iam.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  name: awssqs-sample-source
+spec:
+  annotations:
+    iam.amazonaws.com/role: AWS_IAM_ROLE
+  queueUrl: QUEUE_URL
+  sink:
+    apiVersion: messaging.knative.dev/v1alpha1
+    kind: Channel
+    name: awssqs-test
```
Comment (mattmoor) [inline awssqs/samples/awssqs-source-kube2iam.yaml]: Add trailing newline:
```suggestion
    name: awssqs-test

```

**Comment by lberk** [comment]
Quick nit for the release notes:
#993 #986 and #1012 should all be listed under the 'Fixes'
And #996 #1022 and #1013 under the backports as they're the actual PRs we're backporting

**Comment by AceHack** [comment]
@lberk Updated

**Comment by lberk** [comment]
I don't see the yaml updates? Perhaps the commit hasn't been pushed?

**Comment by AceHack** [comment]
@lberk which yaml files are you talking about?

File: awssqs/config/201-clusterrole.yaml [hunk 2]
```diff
@@ -84,6 +84,14 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - eventing.knative.dev
+  resources:
+  - eventtypes
```
[the remark this answers is not in the document]
Comment (AceHack) [reply awssqs/config/201-clusterrole.yaml]: This has been moved as requested, I believe this should be added to master as well.  I can do that PR.  I added it to the 0.11.x branch and I can add to master.  In my testing on the 0.11.x branch without it there were constant errors in the logs.

File: awssqs/config/201-clusterrole.yaml [hunk 3]
```diff
@@ -92,6 +100,15 @@ rules:
   verbs:
   - get
   - list
+# Services
```
[the remark this answers is not in the document]
Comment (AceHack) [reply awssqs/config/201-clusterrole.yaml]: Updated formatting

[the remark this answers is not in the document]
Comment (lberk) [reply awssqs/config/201-clusterrole.yaml]: Can we fix this one as well?  Comparing it to other source's, it seems perfectly fine to have a `watch` on secrets so we can batch these both under the same `apiGroup`.

[the remark this answers is not in the document]
Comment (lberk) [reply awssqs/config/201-clusterrole.yaml]: @AceHack this isn't resolved.  Please fix as described.

**Comment by AceHack** [comment]
@lberk updated

**Comment by lberk** [comment]
@AceHack hm? I still don't see the `secrets` and `services` under the same `resources.apiGroups` clause?

**Comment by AceHack** [comment]
@lberk sorry I misunderstood the first time, I think I fixed it correctly this time.  Thanks and let me know if you would like any other changes.

**Pull request merged**